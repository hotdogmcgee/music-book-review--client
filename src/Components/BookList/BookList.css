#book-list {
  list-style: none;
  padding: 0 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
}
