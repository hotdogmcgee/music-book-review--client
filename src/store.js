export const STORE = {
    bookList: [
        {
            instrument: 'piano',
            image: 'an image',
            title: 'where the red fern grows',
            author: 'zi forgot',
            description: 'two dogs in a very sad tale',
            rating: 4,
            numReviews: 10,
            id: 1,
            published_year: 2001,
            cost: 30,
            reviews: [
                {
                    user: {
                        full_name: 'dave bones'
                    },
                    rating: 3,
                    id: 1,
                    text: 'it was pretty good'
                }
            ]
        },
        {
            instrument: 'guitar',
            image: 'another image',
            title: 'Harry Potter and the Goblet of Fire',
            author: 'JK Rowling',
            description: 'everyone knows it by now',
            rating: 3,
            numReviews: 3,
            cost: 20,
            published_year: 1990,
            id: 2
        }
    ]
}