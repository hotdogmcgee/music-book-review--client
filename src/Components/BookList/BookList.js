import React from "react";
import { Section } from "../Utils/Utils";
import BookListItem from '../BookListItem/BookListItem'
import './BookList.css'

export default function BookList(props) {
  const displayList = props.bookList
    ? props.bookList.map((item, index) => {
        return (
          <li key={index} >
              <BookListItem {...item}/>
          </li>
        );
      })
    : "error!";

  return (
    <Section>
      <ul id="book-list">{displayList}</ul>
    </Section>
  );
}

