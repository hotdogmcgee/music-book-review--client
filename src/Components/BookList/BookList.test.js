import React from 'react';
import BookList from './BookList.js'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { STORE } from '../../store.js'

describe('BookList component', () => {

    const list = STORE.bookList

      it('renders a BookList by default', () => {
        const wrapper = shallow(<BookList />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    
      it('renders the BookList given props', () => {
        const wrapper = shallow(<BookList bookList={list} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})