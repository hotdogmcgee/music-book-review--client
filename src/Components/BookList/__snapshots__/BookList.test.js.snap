// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component renders a BookList by default 1`] = `
<Section>
  <ul
    id="book-list"
  >
    error!
  </ul>
</Section>
`;

exports[`BookList component renders the BookList given props 1`] = `
<Section>
  <ul
    id="book-list"
  >
    <li
      key="0"
    >
      <BookListItem
        author="zi forgot"
        cost={30}
        description="two dogs in a very sad tale"
        id={1}
        image="image"
        instrument="piano"
        numReviews={10}
        published_year={2001}
        rating={4}
        reviews={
          Array [
            Object {
              "id": 1,
              "rating": 3,
              "text": "it was pretty good",
              "user": Object {
                "full_name": "dave bones",
              },
            },
          ]
        }
        title="where the red fern grows"
      />
    </li>
    <li
      key="1"
    >
      <BookListItem
        author="JK Rowling"
        cost={20}
        description="everyone knows it by now"
        id={2}
        image="image"
        instrument="guitar"
        numReviews={3}
        published_year={1990}
        rating={3}
        title="Harry Potter and the Goblet of Fire"
      />
    </li>
    <li
      key="2"
    >
      <BookListItem
        author="JK Rowling"
        cost={15}
        description="not my fav"
        id={3}
        image="image"
        instrument="guitar"
        numReviews={4}
        published_year={1988}
        rating={2}
        title="Harry Potter and the Chamber of Secrets"
      />
    </li>
    <li
      key="3"
    >
      <BookListItem
        author="Sally Reed"
        cost={30}
        description="the best ever"
        id={4}
        image="image"
        instrument="clarinet"
        numReviews={100}
        published_year={2002}
        rating={5}
        title="The Only Way to Learn Clarinet"
      />
    </li>
  </ul>
</Section>
`;
